core/src/main/java/org/aerogear/mobile/core/configuration/MobileCoreJsonParser.java

```diff
@@ -69,18 +75,42 @@ private void parseConfigObject(final JSONObject jsonObject) throws JSONException
         values.put(serviceConfig.getName(), serviceConfig);
     }
 
+    private void parseHttpsArray(final JSONArray array) throws JSONException, IOException{
+        final int arrayLength = nonNull(array, "jsonArray").length();
+        for(int i=0; i< arrayLength; i++){
+            parseHttpsObject(array.getJSONObject(i));
+        }
+    }
+
+    private void parseHttpsObject(final JSONObject jsonObject) throws JSONException, IOException{
+        nonNull(jsonObject, "jsonObject");
+
+        final HttpsConfiguration.Builder httpsConfigBuilder = HttpsConfiguration.newHashConfiguration();
+        httpsConfigBuilder.setHostName(jsonObject.getString("host"));
+        httpsConfigBuilder.setCertificateHash(jsonObject.getString("certificateHash"));
+
+        final HttpsConfiguration httpsConfiguration = httpsConfigBuilder.build();
+        hashes.put(httpsConfiguration.getHostName(), httpsConfiguration.getCertificateHash());
+    }
+
     /**
      * @param jsonStream a inputStream to for mobile-core.json. Please note that this should be
      *        managed by the calling core. The parser will not close the resource when it is
      *        finished.
      *
-     * @return A map of ServiceConfigs mapped by their name.
      * @throws IOException if reading the stream fails
      * @throws JSONException if the json document is malformed
      */
-    public static Map<String, ServiceConfiguration> parse(final InputStream jsonStream)
+    public static void parse(final InputStream jsonStream)
                     throws IOException, JSONException {
-        MobileCoreJsonParser parser = new MobileCoreJsonParser(jsonStream);
-        return parser.values;
+        new MobileCoreJsonParser(jsonStream);
```

Comment:
See my previous comment: here you must instantiate and throw the object, because this is a malformed parser (no parse method defined)